shacl12-profiling/index.html

```diff
@@ -804,29 +804,106 @@ <h3>Document Conventions</h3>
+      </section>
+      <section id="null-profiles">
+        <h3>Creating Null Profiles</h3>
+        <p>
+          A "null" profile of a specification, created using SHACL, is a profile of that specification in which SHACL is used to implement some, and potentially all the specification's rules, but no other rules, in SHACL.
+        </p>
+        <p>
+          The main purpose of creating a null profile of a specification using SHACL is to enable the testing of conformance to that specification by SHACL validation. This purpose exists because many RDF data models exist that have a model specification, perhaps an OWL model or only a natural-language document of a model, but which do not provide a mechanism for data validation, such as a SHACL validator.
+        </p>
+        <div class="note">
+          <p>
+            While null profiles of specifications can be created using mechanism other than SHACL, here we focus on the use of SHACL only.
+          </p>
+          <p>
+            An example of a non-SHACL validator for RDF data that acts as a null profile is the W3C's [[[prov-constraints]]] which provides a list of constraints that apply to provenance data formulated according to [[[prov-o]]]. [[[prov-constraints]]] implements no constraints beyond those stated or implied in [[[prov-o]]] and the conceptual [[[prov-dm]]], however it does include tests for things that the models do not explicitly model but whose proper use requires, for ordering of temporal entities. Implementations of those constraints <a href="https://github.com/pgroth/prov-check/tree/master">have been made</a> as Python scripts that execute SPARQL allowing for RDF data validation. It would be possible to implement those constraints in SHACL too.
+          </p>
+        </div>
+      </section>
 		</section>
 
 		<section id="syntax-rules" class="appendix">
 			<h2>Summary of Syntax Rules from this Specification</h2>
 		</section>
 
-		<section id="security">
-			<h2>Security Considerations</h2>
-			<p>TODO</p>
-		</section>
-
-		<section id="privacy">
-			<h2>Privacy Considerations</h2>
-			<p>TODO</p>
+		<section id="security" class="appendix informative">
+			<h2>Security and Privacy Considerations</h2>
+			<p>
+				Like most RDF-based technologies, SHACL processors may operate on graphs that are combined
+				from various sources.  Some applications may have an open "linked data" architecture and dynamically
+				assemble RDF triples from sources that are outside an organization's network of trust.
+				Since RDF allows anyone to add statements about any resource, triples may modify the originally
+				intended semantics of shape definitions or nodes in a data graph and thus lead to misleading results.
+				Protection against this (and the following) scenario can be achieved by only using trusted
+				and verified RDF sources and eliminating the possibility that graphs are dynamically added via
+				<code>owl:imports</code> and <code>sh:shapesGraph</code>.
```

Comment:
```suggestion
				Like most RDF-based technologies, SHACL processors may operate on graphs that are assembled
				from various sources.  Some applications may have an open "linked data" ("LD") architecture and dynamically
				assemble RDF triples from sources that are outside an organization's network of trust.
				Since RDF allows anyone to add statements about any resource, triples may modify the originally
				intended semantics of shape definitions or nodes in a data graph and thus feed into misleading results.
				Protection against this (and the following) scenario is achievable by using only trusted
				and verified RDF sources and eliminating the possibility that graphs are dynamically added via
				<code>owl:imports</code> and <code>sh:shapesGraph</code>.
```